Tasks/DotnetCLIRestore/restorecommand.ts

```diff
@@ -0,0 +1,306 @@
+import * as fs from "fs";
+import * as ltx from "ltx";
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as Q from 'q';
+import * as auth from 'packaging-common/nuget/Authentication';
+import { NuGetConfigHelper2 } from 'packaging-common/nuget/NuGetConfigHelper2';
+import * as ngRunner from 'packaging-common/nuget/NuGetToolRunner2';
+import * as path from 'path';
+import { IExecOptions } from 'azure-pipelines-task-lib/toolrunner';
+import * as request from 'request';
+
+export async function run(): Promise<void> {
+    const buildIdentityDisplayName: string = null;
+    const buildIdentityAccount: string = null;
+
+    try {
+        let packageName: string = tl.getInput("packageName");
+        let packageVersion: string = tl.getInput("version") || "";
+
+        if (!packageName)
+        {
+            return;
+        }
+        else {
+            packageName = packageName.split("/")[1];
+        }
+
+        let endpointNames = tl.getDelimitedInput("externalEndpoints", ',');
+        const username: string = await GetGitHubUser(endpointNames[0]); // we will always have a single connection
+
+        const noCache = true;
+        const verbosity = "minimal";
+        let packagesDirectory = tl.getPathInput('packagesDirectory');
+        if (!tl.filePathSupplied('packagesDirectory')) {
```

Comment:
Curious: what does this function do?

Reply:
From the doc: "Checks whether a path inputs value was supplied by the user"